server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -104,7 +105,7 @@ public AggregationProcessor aggregationProcessor(SearchContext searchContext) {
 
     private static <T extends Exception> void rethrowCauseIfPossible(RuntimeException re, SearchContext searchContext) throws T {
         // Rethrow exception if cause is null
-        if (re.getCause() == null) {
+        if (re.getCause() == null || re instanceof OpenSearchException) {
```

Comment:
why is `re.getCause` null here instead shouldn't it return `OpenSearchException` ? From lucene layer, it seems like it will always throw `RuntimeException` with proper `cause` if a task failed to execute. Ref [here](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/TaskExecutor.java#L65)

Reply:
I believe the problem comes from [here](https://github.com/apache/lucene/blob/13e747f95f7b8277cc961e83b4990e6d152dd351/lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java#L665-L668) -- if there is only 1 slice then we just directly call `ContextIndexSearcher::search` and then we directly throw `OpenSearchException`.

Reply:
yes @jed326 is right here